string peer1
string peer2